url89cafed56a_AlexDiru_esper_compiler_final_tgz-pJ8-compiler_ProgramJ8/src/compiler/EsperParser.java

```diff
@@ -0,0 +1,1997 @@
+    	retval.tree = (Object)adaptor.errorNode(input, retval.start, input.LT(-1), re);
+
+        }
+
+        finally {
+        	// do for sure before leaving
+        }
+        return retval;
+    }
+    // $ANTLR end "declaration"
+
+
+    public static class assign_return extends ParserRuleReturnScope {
+        Object tree;
+        public Object getTree() { return tree; }
+    };
+
+
+    // $ANTLR start "assign"
+    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:45:1: assign : ASSIGN ^ IDENTIFIER expr ;
+    public final EsperParser.assign_return assign() throws RecognitionException {
+        EsperParser.assign_return retval = new EsperParser.assign_return();
+        retval.start = input.LT(1);
+
+
+        Object root_0 = null;
+
+        Token ASSIGN24=null;
+        Token IDENTIFIER25=null;
+        EsperParser.expr_return expr26 =null;
+
+
+        Object ASSIGN24_tree=null;
+        Object IDENTIFIER25_tree=null;
+
+        try {
+            // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:45:8: ( ASSIGN ^ IDENTIFIER expr )
+            // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:45:10: ASSIGN ^ IDENTIFIER expr
+            {
+            root_0 = (Object)adaptor.nil();
+
+
+            ASSIGN24=(Token)match(input,ASSIGN,FOLLOW_ASSIGN_in_assign364); 
+            ASSIGN24_tree = 
+            (Object)adaptor.create(ASSIGN24)
+            ;
+            root_0 = (Object)adaptor.becomeRoot(ASSIGN24_tree, root_0);
+
+
+            IDENTIFIER25=(Token)match(input,IDENTIFIER,FOLLOW_IDENTIFIER_in_assign367); 
+            IDENTIFIER25_tree = 
+            (Object)adaptor.create(IDENTIFIER25)
+            ;
+            adaptor.addChild(root_0, IDENTIFIER25_tree);
+
+
+            pushFollow(FOLLOW_expr_in_assign369);
+            expr26=expr();
+
+            state._fsp--;
+
+            adaptor.addChild(root_0, expr26.getTree());
+
+            }
+
+            retval.stop = input.LT(-1);
+
+
+            retval.tree = (Object)adaptor.rulePostProcessing(root_0);
+            adaptor.setTokenBoundaries(retval.tree, retval.start, retval.stop);
+
+        }
+        catch (RecognitionException re) {
+            reportError(re);
+            recover(input,re);
+    	retval.tree = (Object)adaptor.errorNode(input, retval.start, input.LT(-1), re);
+
+        }
+
+        finally {
+        	// do for sure before leaving
+        }
+        return retval;
+    }
+    // $ANTLR end "assign"
+
+
+    public static class ifthenelse_return extends ParserRuleReturnScope {
+        Object tree;
+        public Object getTree() { return tree; }
+    };
+
+
+    // $ANTLR start "ifthenelse"
+    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:46:1: ifthenelse : if_ ( elseif )* ( else_ )* ENDIF ;
+    public final EsperParser.ifthenelse_return ifthenelse() throws RecognitionException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 50 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.